presto-main/src/main/java/com/facebook/presto/testing/TestEquality.java

```diff
@@ -0,0 +1,175 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.facebook.presto.testing;
+
+import com.facebook.presto.spi.block.Block;
+import com.facebook.presto.spi.type.DoubleType;
+import com.facebook.presto.spi.type.Type;
+import com.facebook.presto.type.ArrayType;
+import com.facebook.presto.type.MapType;
+import com.facebook.presto.type.RowType;
+
+import java.util.HashMap;
+import java.util.Map;
+
+import static com.facebook.presto.testing.TestingConnectorSession.SESSION;
+import static com.facebook.presto.type.TypeUtils.checkElementNotNull;
+import static java.lang.Double.isInfinite;
+import static java.lang.Double.isNaN;
+
+public final class TestEquality
+{
+    private TestEquality() {}
+
+    public static boolean equalToForTests(Type type, Block leftBlock, int leftPosition, Block rightBlock, int rightPosition)
+    {
```

Comment:
This is code copied from a bunch of type.equalTo methods but is a little more permissive because tests depend on stuff like NaNs comparing intuitively.